packages/SwingSet/src/liveslots/liveslots.js

```diff
@@ -905,6 +923,14 @@ function build(
     const maybeNewVPIDs = new Set(serMethargs.slots);
     const resolutions = resolutionCollector().forSlots(serMethargs.slots);
     if (resolutions.length > 0) {
+      try {
+        resolutions.forEach(([_xvpid, _isReject, resolutionCD]) => {
+          assertAcceptableSyscallCapdataSize(resolutionCD);
+        });
+      } catch (e) {
+        syscall.exit(true, e);
```

Comment:
hm, I wonder if we can give them any extra hints here, a `console.log` of something, so the author can figure out why their vat just suddenly self-destructs

The VPID might be a clue for someone looking deeply enough at the slogs. Some prefix of the `resolutionCD` might be a hint. But neither one should really be revealed to userspace, so we'd have to `console.log` it rather than making the assertion function include the hint in the Error object (which is visible to the `send()` callers). And unless it's in the Error object, this `forEach` pattern would lose access to the hint by the time it reaches the `syscall.exit`.

Hrmph. Well, I guess the slogs will still show the last delivery, and any previous syscalls that were successful. And maybe a replay of the fatal delivery might let them learn more. Still, let's keep thinking about ways to make this failure mode more ergonomic.

Reply:
I'm slightly less concerned about this since, realistically, I expect this error case to arise very rarely if ever.